src/agenteval/cli.py

```diff
@@ -371,44 +311,113 @@ def publish_command(
-    eval_result.submission.agent_url = agent_url
-    eval_result.submission.submit_time = datetime.now(timezone.utc)
-    eval_result.submission.openness = openness
-    eval_result.submission.tool_usage = tool_usage
+    submission = SubmissionMetadata(
+        username=username,
+        agent_name=agent_name,
+        agent_description=agent_description,
+        agent_url=agent_url,
+        submit_time=datetime.now(timezone.utc),
+        openness=openness,
+        tool_usage=tool_usage,
+    )
+
+    atomic_write_file(
+        os.path.join(log_dir, SUBMISSION_METADATA_FILENAME),
+        submission.model_dump_json(indent=2),
+    )
 
     # Validate suite config version
-    config_name = eval_result.suite_config.version
+    config_name = eval_config.suite_config.version
     if not config_name:
         raise click.ClickException("Suite config version is required for upload.")
 
     # Build submission name
-    ts = eval_result.submission.submit_time.strftime("%Y-%m-%dT%H-%M-%S")
-    subm_name = f"{safe_username}_{safe_agent_name}_{ts}"
+    ts = submission.submit_time.strftime("%Y-%m-%dT%H-%M-%S")
+    submission_name = f"{safe_username}_{safe_agent_name}_{ts}"
 
     # Upload logs and summary
     logs_url = upload_folder_to_hf(
-        hf_api, log_dir, submissions_repo_id, config_name, eval_result.split, subm_name
+        hf_api,
+        log_dir,
+        submissions_repo_id,
+        config_name,
+        eval_config.split,
+        submission_name,
     )
     click.echo(f"Uploaded submission logs dir to {logs_url}")
-    eval_result.submission.logs_url = logs_url
 
+
+cli.add_command(publish_logs_command)
+
+
+@click.command(
+    name="publish",
+    help="Publish scored results in log_dir to HuggingFace leaderboard.",
+)
+@click.argument("submission_url", type=str)
+@click.option(
+    "--repo-id",
+    default="allenai/asta-bench-internal-results",
+    required=False,
+    help="HuggingFace repo",
+)
+def publish_lb_command(repo_id: str, submission_url: str):
+    hf_url_match = re.match(HF_URL_PATTERN, submission_url)
+    if not hf_url_match:
+        click.echo(
+            f"Invalid submission URL format: {submission_url}. "
+            "Expected format: hf://<repo_id>/<submission_path>"
+        )
+        sys.exit(1)
+    repo_id = hf_url_match.group(1)
+    submission_path = hf_url_match.group(2)
+
+    from huggingface_hub import HfApi, snapshot_download
+
+    hf_api = HfApi()
+
+    submission_summary_dir = snapshot_download(
+        repo_id=repo_id,
+        allow_patterns=[
+            f"{submission_path}/{EVAL_CONFIG_FILENAME}",
+            f"{SUMMARIES_PREFIX}/{submission_path}/{SCORES_FILENAME}",
+            f"{submission_path}/{SUBMISSION_METADATA_FILENAME}",
+        ],
+    )
+    eval_config_path = scores_path = os.path.join(
+        submission_summary_dir, EVAL_CONFIG_FILENAME
+    )
+    scores_path = os.path.join(submission_summary_dir, SCORES_FILENAME)
```

Comment:
wouldn't the scores file be under a different subdir from the eval config file and the submission metadata file? since they go under the summaries prefix? (though maybe I'm just not familiar enough with snapshot_download()?)